src/memberAPI.ts

```diff
@@ -1,70 +1,163 @@
+    let member = await (
+      await db.doc('members/' + req.session.email).get()
+    ).data();
     if (!member) {
-      res.status(200).json({ error: "Not member with email: " + req.session.email });
+      return {
+        status: 404,
+        error: 'No member with email: ' + req.session.email
+      };
     } else {
       let canEdit = PermissionsManager.canEditMembers(member.role);
-      if (!canEdit) {
-        res.status(200).json({ error: "User with email: " + req.session.email + " does not have permission to edit members!" });
+      if (!canEdit && member.email !== req.body.email) {
+        // members are able to edit their own information
+        return {
+          status: 403,
+          error: 'User with email: ' + req.session.email + ' does not have permission to edit members!'
+        };
       } else {
         if (!req.body.email || req.body.email === '') {
-          res.status(200).json({ error: "Couldn't delete user with undefined email!" });
-          return;
+          return {
+            status: 400,
+            error: "Couldn't edit user with undefined email!"
+          };
         }
-        db.doc('members/' + req.body.email).delete().then(() => {
-          res.status(200).json({ status: "Success", member: req.body });
-        }).catch((reason) => {
-          res.status(200).json({ error: "Couldn't delete user for reason: " + reason });
-        });
+        if ((req.body.role || req.body.first_name || req.body.last_name) &&
+          !canEdit) {
+          return {
+            status: 403,
+            error: 'User with email: ' + req.session.email + ' does not have permission to edit member name or roles!',
+          };
+        }
+        db.doc('members/' + req.body.email)
+          .update(req.body)
+          .then(() => {
+            return {
+              status: 200,
+              member: req.body
+            };
+          })
+          .catch((reason) => {
+            return {
+              status: 500,
+              error: "Couldn't edit user for reason: " + reason
+            };
+          });
+      }
+    }
+  }
+};
+
+export let getMember = async function (req: Request, res: Response): Promise<MemberResponse | ErrorResponse> {
+  if (checkLoggedIn(req, res)) {
+    let member = await (
+      await db.doc('members/' + req.session.email).get()
+    ).data();
+    console.log(member);
+    if (!member) {
+      return {
+        status: 404,
+        error: 'No member with email:' + req.session.email
+      };
+    } else {
+      let canEdit: boolean = PermissionsManager.canEditMembers(member.role);
+      let memberEmail: string = req.params.email;
+      if (!canEdit && memberEmail !== req.session.email) {
+        return {
+          status: 403,
+          error: 'User with email: ' + req.session.email + ' does not have permission to get members!'
+        };
       }
+      let getmember: Member = (await db.doc('members/' + memberEmail).get()).data() as Member;
```

Comment:
This variable name could probably just be `member`. `getmember` seems a bit confusing to me.